internal/olm/operator/bundle/install.go

```diff
@@ -98,3 +168,169 @@ func (i *Install) setup(ctx context.Context) error {
 
 	return nil
 }
+
+func (f *FBCContext) generateFBCContent(ctx context.Context, bundleImage, indexImage string) (string, error) {
+	log.Infof("Creating a File-Based Catalog of the bundle %q", bundleImage)
+	// generate a File-Based Catalog representation of the bundle image
+	bundleDeclcfg, err := f.createFBC(ctx)
+	if err != nil {
+		return "", fmt.Errorf("error creating a File-Based Catalog with image %q: %v", bundleImage, err)
+	}
+
+	declcfg := &declarativeconfig.DeclarativeConfig{
+		Bundles:  []declarativeconfig.Bundle{bundleDeclcfg.Bundle},
+		Packages: []declarativeconfig.Package{bundleDeclcfg.Package},
+		Channels: []declarativeconfig.Channel{bundleDeclcfg.Channel},
+	}
```

Comment:
I may have missed this in the review. Why are we creating a new object with the same values as the original `bundleDeclcfg`?

Reply:
Coming back to this. We created the unexported struct to remove all checks of != 1. Isn't that what needs to be used here?

`createFBC` will return a "minimal" FBC (with one in each field), and so we'd create the `bundleDeclcfg` object here and propagate it further, correct?

Reply:
Disregard, I just saw that the createFBC return type changed since I last reviewed it. This is fine.

Reply:
@VenkatRamaraju `createFBC` will return a minimal FBC that is of the `bundleDeclcfg` type. I'm creating `declcfg` to convert the `bundleDeclcfg` to the more `declarativeconfig.DeclarativeConfig` type as `addBundleToIndexImage` function expects that type. 

I think the idea here was to create this unexported struct to get rid of the length checks. We are creating exactly one element of each type i.e bundles/channels/packages during `createFBC` invocation and then we need to convert this to the `declarativeconfig.DeclarativeConfig` type before `addBundleToIndexImage` so we know that it has exactly one element of each of those.